lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +678,143 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        final int wsid = ev.getAccessGroupId().get();
+        final long objid = Long.valueOf(ev.getAccessGroupObjectId().get().toString()).longValue();
+
+        Map<String, Object> command;
+
+        // check if workspace is permanently deleted or marked as deleted
+        try {
+            final Tuple9 tuple = getWorkspaceInfo(wsid);
+        } catch(IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch(JsonClientException ex) {
+            // hacky: if exception message is changed, this logic will silently fail
+            if (ex.getMessage().contains("No workspace with id ")) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                StatusEventType.DELETE_ALL_VERSIONS).
+                        withNullableAccessGroupID(wsid).
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        }
+
+        // check if object is permanently deleted or marked as deleted
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().
+                                         withIds(Arrays.asList((long)wsid)).
+                                         withMinObjectID(objid-1).
+                                         withMaxObjectID(objid+1));
+        try {
+            boolean objidExists = false;
+
+            final List<List> objList = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(List.class);
+
+            for (List obj : objList) {
+                final long id = (long) ((Integer) obj.get(0)).intValue();
+                if (id == objid) {
+                    objidExists = true;
+                    break;
+                }
+            }
+
+            if (!objidExists) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                StatusEventType.DELETE_ALL_VERSIONS).
+                        withNullableAccessGroupID(wsid).
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        } catch (IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch (JsonClientException ex) {
+            // really think it should be IndexingException here
+            // but IndexingException has a protected constructor
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        }
+
+        // check if name has changed and update state of lastestName
+        ObjectSpecification os = new ObjectSpecification().withRef(
```

Comment:
Seems like all this duplicates the getObjectInfo() method above?

Reply:
reused getObjectInfo()